pkg/cri/opts/spec_linux.go

```diff
@@ -172,6 +174,19 @@ func WithCDI(annotations map[string]string, CDIDevices []*runtime.CDIDevice) oci
 			log.G(ctx).Debug("Passing CDI devices as annotations will be deprecated soon, please use CRI CDIDevices instead")
 		}
 
-		return oci.WithCDIDevices(devices...)(ctx, client, c, s)
+		for _, mount := range CDIMounts {
```

Comment:
If this is all that's getting done, why don't we "just" create a CDI spec for a unique device here and then consume the CDI specification normally?

If that is not done here, who is responsible for generating this specification?